eng/common/pipelines/templates/steps/update-docsms-metadata.yml

```diff
@@ -0,0 +1,103 @@
+parameters:
+  - name: Artifacts
+    type: object
+    default: []
+  - name: RepoId
+    type: string
+    default: $(Build.Repository.Name)
+  - name: WorkingDirectory
+    type: string
+    default: ''
+  - name: ScriptDirectory
+    type: string
+    default: eng/common/scripts
+  - name: TargetDocRepoName
+    type: string
+    default: ''
+  - name: TargetDocRepoOwner
+    type: string
+  - name: Language
+    type: string
+    default: ''
+  - name: DailyDocsBuild
+    type: boolean
+    default: false
+  - name: SparseCheckoutPaths
+    type: object
+    default:
+      - '**'
+
+steps:
+- template: /eng/common/pipelines/templates/steps/enable-long-path-support.yml
+
+- pwsh: |
+    Write-Host "###vso[task.setvariable variable=DocRepoLocation]${{ parameters.WorkingDirectory }}/doc"
+  displayName: Set $(DocRepoLocation)
+
+- template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+  parameters:
+    SkipDefaultCheckout: true
+    Repositories:
+      - Name: ${{ parameters.TargetDocRepoOwner }}/${{ parameters.TargetDocRepoName }}
+        WorkingDirectory: $(DocRepoLocation)
+    Paths: ${{ parameters.SparseCheckoutPaths }}
+
+# If performing a daily docs build set the $(TargetBranchName) to a daily branch
+# name and attempt to checkout the daily docs branch. If the branch doesn't
+# exist, create it
+- ${{ if eq(parameters.DailyDocsBuild, 'true') }}:
+  - template: /eng/common/pipelines/templates/steps/set-daily-docs-branch-name.yml
+    parameters:
+      DailyBranchVariableName: TargetBranchName
+
+  - pwsh: |
+      $ErrorActionPreference = "Continue"
+      $RemoteName = "origin"
+      $BranchName = "$(TargetBranchName)"
+      # Fetch and checkout remote branch if it already exists otherwise create a new branch.
+      git ls-remote --exit-code --heads $RemoteName $BranchName
+      if ($LASTEXITCODE -eq 0) {
```

Comment:
Looks like you found a good solution. Should I close https://github.com/Azure/azure-sdk-tools/issues/1642 ?

Reply:
We've run into this problem in other parts of the codebase. @weshaggard has designs to pull this logic into a yaml template of sorts. 

I think valid arguments can be made for including this in sparse-checkout.yml (it's easier to do everything in one template call as opposed to two) or not (call the second template when you need it, don't bloat `sparse-checkout.yml`).  

One deciding factor could be whether we would use this kind of branch logic in a context outside of a sparse checkout?

Reply:
Is what you're suggesting basically a "checkout main and then make new branch" option?

```
- template: sparse-checkout.yml
  parameters:
    NewBranch: <branch name>
```

We would also need to consider whether to support this parameter as global to the template and/or specific to each Repositories object.

Reply:
In the case where the branch does not exist, yes, check out the default branch and then create the new branch. However the branch may already exist and in that case we want to just check out that branch.